### PR TITLE
feat: add express error handler middleware

### PR DESCRIPTION
**Pull request by VitaliiDenysiuk**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -13,6 +14,8 @@ app.use(express.json());
 
 app.use("/api", apiRoutes);
 
+app.use(invalidPathHandler);
```
Comment (alexandrtovmach) [inline src/index.ts]: it's not a "middleware" and it's not a "path" + what code should be used for not found?

Comment (alexandrtovmach) [reply src/index.ts]: https://stackoverflow.com/questions/6528876/how-to-redirect-404-errors-to-a-page-in-expressjs
```
app.use(function(req, res, next) {
  res.status(404);

  if (req.accepts('json')) {
    res.json({ error: 'Not found' });
    return;
  }

  res.type('txt').send('Not found');
});
```

File: src/middlewares/invalidPathHandler.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { Request, Response } from "express";
```
Comment (alexandrtovmach) [inline src/middlewares/invalidPathHandler.ts]: delete file

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]
implemented wrong thing
- create general error handler that catch errors on global level
- parse error message
- send parsed error message to response
- logging full error message

File: src/index.ts (hunk 1 above)
Comment (VitaliiDenysiuk) [reply src/index.ts]: done

File: src/middlewares/invalidPathHandler.ts (hunk 2 above)
Comment (VitaliiDenysiuk) [reply src/middlewares/invalidPathHandler.ts]: done

**Comment by VitaliiDenysiuk** [comment]
> * create general error handler that catch errors on global level
> * parse error message
> * send parsed error message to response
> * logging full error message

@alexandrtovmach at that moment nothing to parse.
I use additional article for doing that task
[https://www.codeconcisely.com/posts/how-to-handle-errors-in-express-with-typescript/](url)
[https://www.becomebetterprogrammer.com/how-to-use-error-handler-middleware-with-express-js-and-typescript/](url)

now in all parts of project you can call "throw error" and it comes to errorHandler.middlewares.ts
and all mistaces go to log

**Comment by VitaliiDenysiuk** [comment]
@alexandrtovmach   if I don't find something tell me

**Review by alexandrtovmach — APPROVED** [review]

**Review by fominmaksim — APPROVED** [review]

**Pull request merged**